# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/GCMarker.h
# Commit: 6e5a935e576a
# Full Hash: 6e5a935e576a5263c85c48b7c39450464b2b047e
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-10-28 21:22:11
# Regressor Bug: 1797755
# File Overlap Count: 1
# Description:
#   Bug 1797755 - Part 3: Remove the unused MarkStackIter class r=sfink
#   
#   Depends on D160525
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D160526
# ==============================================================================

diff -r eb109d0ce408 -r 6e5a935e576a js/src/gc/GCMarker.h
--- a/js/src/gc/GCMarker.h	Fri Oct 28 09:47:14 2022 +0000
+++ b/js/src/gc/GCMarker.h	Fri Oct 28 09:47:14 2022 +0000
@@ -204,28 +204,6 @@
 #ifdef DEBUG
   mutable size_t iteratorCount_;
 #endif
-
-  friend class MarkStackIter;
-};
-
-class MarkStackIter {
-  MarkStack& stack_;
-  size_t pos_;
-
- public:
-  explicit MarkStackIter(MarkStack& stack);
-  ~MarkStackIter();
-
-  bool done() const;
-  MarkStack::Tag peekTag() const;
-  MarkStack::TaggedPtr peekPtr() const;
-  MarkStack::SlotsOrElementsRange peekSlotsOrElementsRange() const;
-  void next();
-  void nextPtr();
-  void nextArray();
-
- private:
-  size_t position() const;
 };
 
 } /* namespace gc */